model/compile/QtiAssetCompiler/QtiItemAssetCompiler.php

```diff
@@ -100,9 +100,10 @@ private function resolve(ItemMediaResolver $resolver, string $assetUrl, string $
         return new PackedAsset($type, $mediaAsset, $link);
     }
 
-    private function getReplacementName(string $link): string
+    private function getReplacementName(PackedAsset $packedAsset): string
     {
-        return hash('crc32', basename($link));
+        $link = $packedAsset->getMediaAsset()->getMediaIdentifier();
+        return $packedAsset->getMediaAsset()->getMediaSource()->getBaseName($link);
```

Comment:
Is it covered in the test? I remember this was a critical part changed before and we had some issues. If possible would be nice to have a test for this.